[ATfL] Add changelog (#271)
arm/arm-toolchain: arm-software/linux/CHANGELOG.md

# Changelog

All notable changes to this project will be documented in this file.

## [20.1.0]

This is the first release of the Arm Toolchain for Linux (ATfL), a successor of
the Arm Compiler for Linux.

> Although it is entirely based on the LLVM project, there were some changes

Reviewer: Do we want to mention at least which main release we're basing this on (maybe it's covered elsewhere, so not required?)

introduced specifically for this release. Most notably, these were:

- The compiler uses a config file by default, which improves
  performance-specific optimizations; most notably, it encourages the use of the

>   vectorized math routines in the Loop Vectorizer which enables possibility of

Reviewer: "enables the possibility of"?

  vectorizing loops containing the calls to the mathematical library functions.

- For function whose `vscale_range` is limited to a single value, ATfL can size
  scalable vectors.
  See [this pull request](https://github.com/llvm/llvm-project/pull/130973) for
  more details.

- A part of transformation in the Loop Vectorizer causing 'Verification Error'
  on the WRF benchmark has been deactivated.
  See [this bug report](https://github.com/llvm/llvm-project/issues/126836) for
  more details.

> - The Bash autocompletion covers `armclang`, `armclang++` and `armflang` too.

Reviewer: "too" here refers to what?

Author: Done

# Changelog

All notable changes to this project will be documented in this file.

## [20.1.0]

This is the first release of the Arm Toolchain for Linux (ATfL), a successor of

> the Arm Compiler for Linux.

Reviewer 2: ```suggestion
the Arm Compiler for Linux (ACfL).
```

Author: Done

> Although it is entirely based on the LLVM project version 20.1, there were some
> changes introduced specifically for the ATfL. Most notably, these were:

Reviewer 2: ```suggestion
Although ATfL is based entirely on LLVM version 20.1, several changes have been introduced specifically for this toolchain. The most notable include:
```

Author: Done

- The compiler uses a config file by default, which improves
  performance-specific optimizations; most notably, it encourages the use of the
  vectorized math routines in the Loop Vectorizer which enables the possibility

>   of vectorizing loops containing the calls to the mathematical library

Reviewer: Argh. I thought I wrote a comment here about "math" ans "mathematical". Somehow, I can't find it in the "comments". It isn't important, but it feels a bit jarring. Can we either use math or mathematical in both of the above two places?

Author: Done

  library functions.

- For function whose `vscale_range` is limited to a single value, ATfL can size

>   scalable vectors.

Reviewer 2: Should we add something about permitting bitcast between scalable and fixed.
```
Enable LLVM to perform bitcast-like operations between fixed and scalable vectors, improving optimization opportunities for code utilizing scalable vector types
```

Author: Added an item

Reviewer 2: I think it should be part of this item.

Author: Integrated into one.
